cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/hbase/AbstractHBaseDataSetAdmin.java

```diff
@@ -58,14 +58,14 @@ public String apply(Class<?> input) {
     }
   };
 
-  protected final String tableName;
+  protected final TableId tableId;
   protected final Configuration hConf;
   protected final HBaseTableUtil tableUtil;
 
   private HBaseAdmin admin;
 
   protected AbstractHBaseDataSetAdmin(String name, Configuration hConf, HBaseTableUtil tableUtil) {
-    this.tableName = name;
+    this.tableId = TableId.from(name);
```

Comment:
hm do you think it makes sense to pass in a TableId instead of a string? That way if it is invalid it exposed earlier up the hierarchy.